src/main/kotlin/inspections/variable/ProbabilitiesStorage.kt

```diff
@@ -0,0 +1,70 @@
+package inspections.variable
+
+import com.intellij.psi.PsiNameIdentifierOwner
+import com.intellij.psi.SmartPointerManager
+import com.intellij.psi.SmartPsiElementPointer
+
+class ProbabilitiesStorage {
```

Comment:
Singletons are evil. In this case, the class used as an infinite cache, therefore it may live for too long and consume too much memory. Are we sure that we need this cache? Perhaps, everything will be fine even without it or the issue may be fixed by optimizing somewhere else? 

If we still need it then we can make it more granular (for example per PSI file).